opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -2881,14 +2881,14 @@ namespace Opm
         }
 
         const double pressure_tolerance = param_.tolerance_pressure_ms_wells_;
-        if (residuals[SPres] > pressure_tolerance) {
-            sum += residuals[SPres] / pressure_tolerance;
+        if (residuals[SPres]*scalingFactorPressure > pressure_tolerance) {
+            sum += residuals[SPres] / pressure_tolerance * scalingFactorPressure;
             ++count;
         }
 
         const double control_tolerance = getControlTolerance(well_state, deferred_logger);
-        if (residuals[SPres + 1] > control_tolerance) {
-            sum += residuals[SPres + 1] / control_tolerance;
+        if (residuals[SPres + 1]*scalingFactorPressure > control_tolerance) {
```

Comment:
This part makes me skeptical: It looks like we scale the control equation as if it is a pressure equation, also for the case when it is a rate equation? Maybe getControlTolerance() is a better place to handle that?

Reply:
The scaling of the control equation was not intended. I will remove it and update the PR.